File: 219.存在重复元素-ii.js
/*
 * @lc app=leetcode.cn id=219 lang=javascript
 *
 * [219] 存在重复元素 II
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(nums, k) {
    let idxs = new Map();
    for (let i = 0; i < nums.length; i++) {
        if (idxs.has(nums[i])) {
            if (i - idxs.get(nums[i])<= k) {
                return true;
            }
        }
        idxs.set(nums[i], i);
    }
    return false;
};
// @lc code=end

File: 219.存在重复元素-ii.go
/*
 * @lc app=leetcode.cn id=219 lang=golang
 *
 * [219] 存在重复元素 II
 */

// @lc code=start
func containsNearbyDuplicate(nums []int, k int) bool {
    idxs := make(map[int]int)
	for i, num := range nums {
		if lastNumIdx, ok := idxs[num]; ok {
			if i - lastNumIdx <= k {
				return true
			}
		}
		idxs[num] = i
	}
	return false
}
// @lc code=end

